metapool/prep.py

```diff
@@ -842,3 +846,62 @@ def qiita_scrub_name(name):
         the sample name, formatted for qiita
     """
     return re.sub(r'[^0-9a-zA-Z\-\.]+', '.', name)
+
+
+def _demux_by_project(frame):
+    # use PlateReplication object to convert each sample's 384 well location
+    # into a 96-well location + quadrant. Since replication is performed at
+    # the plate-level, this will identify which replicants belong in which
+    # new sample-sheet.
+    plate = PlateReplication(None)
+
+    frame['quad'] = [plate.get_96_well_location_and_quadrant(x)[0] for x in
+                     list(frame.destination_well_id)]
+
+    gb = frame.groupby('quad')
+
+    return [gb.get_group(grp).drop(['quad'], axis=1) for grp in gb.groups]
+
+
+def demux_pre_prep(pre_prep):
+    """Given a pre-prep file w/samples that are plate-replicates, generate new
+       pre-prep files for each unique plate-replicate.
+
+    Parameters
+    ----------
+    pre_prep: A Pandas dataframe representing a pre-prep file.
+        Object from where to extract the data.
+
+    Returns
+    -------
+    list of pre_preps
+    """
+    projects = []
+    data = {}
+
+    for project_name, project in pre_prep.groupby('project_name'):
+        # assert that all values in 'contains_replicates' column for a given
+        # project are either all True or all False. Handle case-variations.
+        project.contains_replicates = project.contains_replicates.str.lower()
+        values = project.contains_replicates.unique()
+
+        if len(values) != 1:
+            raise ValueError(
+                f"values in 'contains_replicates' column contain "
+                "values for both True and False for project "
+                f"'{project_name}'")
+
+        projects.append((project_name, values[0] == 'true'))
+        data[project_name] = project
+
+    results = []
+
+    for project_name, needs_demuxing in projects:
```

Comment:
The above for loop creates a list that then is loop here; why not join the 2 for loops into a single one? Not sure if there is an advantage to loop things twice.